### ISSUE TITLE
Add possibility to add custom loss function to BasePretrainNetwork. And a question

### ISSUE THREAD

**Issue by vlovsky** [deeplearning4j/deeplearning4j#3051]
It seems that I can't configure my own loss function in the BasePretrainNetwork and its descendant RBM. 

Also, I have a general question. It's pretty inconsistent how loss functions are defined and used in different types of networks. Is it possible to introduce a getter on a more generic level? For example in https://github.com/deeplearning4j/deeplearning4j/blob/1f8af820c29cc5567a2c5eaa290f094c4d1492a7/deeplearning4j-nn/src/main/java/org/deeplearning4j/nn/conf/layers/Layer.java

**Comment by eraly** [comment]
So this is part of the larger issue on RBMs and deep belief nets. I would prefer to make sure RBMs and deep belief nets work the way they should before supporting custom loss functions for base pretrain networks. 

The other option is to explore variational auto encoders.

**Comment by vlovsky** [comment]
I didn't find how score even matters (at least in stochastic gradient descent) with respect to RBM learning. Am I missing something or it supposed to be this way?

**Comment by eraly** [comment]
Yes, the score doesn't matter. I don't think the score should even be set when pretraining. The score from a loss function is for supervised training and pretraining RBMs is unsupervised. This is what I saying in the other issue.

**Comment by eraly** [comment]
We are dropping support for RBMs in favour of newer and more effective methods. Please take a look at variational autoencoders 
https://github.com/deeplearning4j/dl4j-examples/tree/master/dl4j-examples/src/main/java/org/deeplearning4j/examples/unsupervised/variational
